pycaret/internal/utils.py

```diff
@@ -662,3 +666,75 @@ def infer_ml_usecase(y: pd.Series) -> Tuple[str, str]:
     else:
         subcase = "binary"
     return ml_usecase, subcase
+
+
+# pycaret report changes
+def global_create_report(model, report_name, export_as,model_type):
+
+    # create the markdown report
+    create_markdown_report(model, report_name,model_type)
+
+    if export_as == 'html':
+        input_filename = report_name + '.md'
+        output_filename = report_name + '.html'
+        with open(input_filename, 'r') as f:
+            html_text = markdown(f.read(), extensions=['fenced_code', 'codehilite'])
+            file = open(output_filename, "w")
+            file.write(html_text)
+            file.close()
+
+
+def create_markdown_report(model, report_name,model_type):
```

Comment:
What happens when a user-defined model is passed?

Reply:
Add support for models not supported by PyCaret